src/jarabe/view/launcher.py

```diff
@@ -155,16 +155,20 @@ def __launch_failed_cb(home_model, home_activity):
     if launcher is None:
         logging.error('Launcher for %s is missing', activity_id)
     else:
-        launcher.error_text.props.label = _('<b>%s</b> failed to start.') % \
-            home_activity.get_activity_name()
+        if hasattr(home_activity, 'is_python2_activity') and home_activity.is_python2_activity:
+            launcher.error_text.props.label = _('<b>%s</b> cannot start because it was made for an older version.') % \
```

Comment:
A better error message would be specifying that it was made for Python2 and that's old.